core/src/crds_gossip_push.rs

```diff
@@ -249,11 +254,14 @@ impl CrdsGossipPush {
                         trace!("new_push_messages insert {} {:?}", *p, v);
                         push_messages.entry(*p).or_default().push(v.clone());
                         self.num_pushes += 1;
+                        *self.last_pushed_to.entry(*p).or_insert(0) = now;
                     }
                 }
                 self.push_messages.remove(&v.label());
             }
         }
+        self.last_pushed_to
+            .retain(|_id, timestamp| now - *timestamp > MAX_PUSHED_TO_TIMEOUT_MS);
```

Comment:
maybe we can just use an lru cache here

Reply:
yea.. is it worth the extra work? I guess it's harder to DOS and has a fixed memory footprint.

Reply:
i think it would also save us from iterating over the entire hashmap here

Reply:
Wouldn't you need to search the map for the highest item to reject when you hit capacity?

Reply:
There are O(1) implementations: https://docs.rs/lru/0.6.0/lru/
We can also switch to lru later. Probably will not make much difference for now.